addon/components/base/bs-button.js

```diff
@@ -449,4 +453,37 @@ export default class Button extends Component {
 
     return this.get('bubble');
   }
+
+  init() {
+    super.init(...arguments);
+
+    // deprecate arguments used for attribute bindings only
+    [
+      ['buttonType:type', 'submit'],
+      ['title', 'foo'],
+      ['disabled', true],
+    ].forEach(([mapping, value]) => {
+      let argument = mapping.split(':')[0];
+      let attribute = mapping.includes(':') ? mapping.split(':')[1] : argument;
+      let deprecationMessage =
+        `Argument ${argument} of <BsButton> component is deprecated. It's only purpose ` +
+        `was setting the HTML attribute ${attribute} of the control element. You should use ` +
+        `angle bracket  component invocation syntax instead:\n` +
+        `Before:\n` +
+        `  {{bs-button ${attribute}=${typeof value === 'string' ? `"${value}"` : value}}}\n` +
+        `  <BsButton @${attribute}=${typeof value === 'string' ? `"${value}"`: `{{${value}}}`} />\n` +
+        `After:\n` +
+        `  <BsButton ${typeof value === 'boolean' ? ( value ? attribute : '' ) : `${attribute}="${value}"`} />`;
+
+      deprecate(
+        deprecationMessage,
+        // eslint-disable-next-line ember/no-attrs-in-components
+        !Object.keys(this.attrs).includes(argument),
```

Comment:
`this.attrs` is internal API and there is even an ESLint rule to not use it. But there isn't a better API available. Relying on it's keys only is actually a recommended interim solution until `@glimmer/components` introduces `this.args` public API. Pzuraq [calls it an _intimate API_](https://discuss.emberjs.com/t/best-way-to-get-attrs-only/17021). I think it's quite unlikely that it will be removed before `@ember/component` is removed entirely.